base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs

```diff
@@ -581,15 +581,16 @@ impl UnconfirmedPool {
                 total_fees += total_tx_fee;
                 total_weight += weight;
             }
-            if last_count > 0 {
-                let stat = FeePerGramStat {
-                    order: start as u64,
-                    min_fee_per_gram,
-                    avg_fee_per_gram: total_fees / total_weight,
-                    max_fee_per_gram,
-                };
-                stats.push(stat);
+            if last_count == 0 {
+                break;
             }
+            let stat = FeePerGramStat {
+                order: start as u64,
+                min_fee_per_gram,
+                avg_fee_per_gram: total_fees / total_weight,
+                max_fee_per_gram,
+            };
+            stats.push(stat);
             offset = last_count;
```

Comment:
Should this not increase?
This will only be able to count 2 weights worth of blocks deep.

Reply:
Does it not increase at line 577?

Reply:
Yeah sounds right, looks like I was a bit rushed with this one ;)

Reply:
I did refactor the code. I got rid of the last_count and enumerate as well. Please take a look.